apisix/plugins/basic-auth.lua

```diff
@@ -80,12 +80,19 @@ local function extract_auth_header(authorization)
             return nil, err
         end
 
+        if not m then
+            return nil, "Invalid authorization header format"
+        end
+
         local decoded = ngx.decode_base64(m[1])
```

Comment:
@spacewander seems decode_base64 always return a `ffi_string`
https://github.com/openresty/lua-resty-core/blob/985eb5b323468effaa66deb5cbd4f800b99834cf/lib/resty/core/base64.lua#L60

Reply:
`ffi_string` is not a type, but a function returns a Lua string

Reply:
Oh, I know what you mean. No, you point to the wrong method. See https://github.com/openresty/lua-resty-core/blob/985eb5b323468effaa66deb5cbd4f800b99834cf/lib/resty/core/base64.lua#L107